server/stream.go

```diff
@@ -2856,7 +2879,20 @@ func (mset *stream) processInboundSourceMsg(si *sourceInfo, m *inMsg) bool {
 // Generate a new style source header.
 func (si *sourceInfo) genSourceHeader(reply string) string {
 	var b strings.Builder
-	b.WriteString(si.iname)
+	// strip the source index number from the iname
+	name := si.iname
+	var index int = -1
+	for i, c := range si.iname {
```

Comment:
If only one there is a function for that, so this could be simplified.

Reply:
There could be one or two ':' in the iname but should always be at least one, I went with writing it this way as I thought since it's invoked for every sourced message it would be a tiny bit quicker than using strings.Cut (if that's the function you mean). Would you rather I simplify the code using Cut?

Reply:
Or just index to find where it is.

https://pkg.go.dev/strings#Index

Reply:
Done using Index